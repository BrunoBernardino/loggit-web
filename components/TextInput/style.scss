@import '__variables';

$transition-speed: 140ms;

.TextInput {
  &__label {
    color: $color-text;
    padding: 0.5rem 1rem 0.5rem 0.7rem;
    border-left: 5px solid $color-block-border;
    text-transform: uppercase;
    font-weight: 500;
    display: block;
    font-size: 1.3rem;

    @media (prefers-color-scheme: dark) {
      color: $color-link-hover;
    }
  }

  &__input {
    color: $color-text-gray;
    font-size: 1.7rem;
    margin: 0.5rem 0 1rem;
    display: block;
    text-align: left;
    padding: 0.7rem 1.5rem;
    border-radius: 5px;
    font-weight: 500;
    background: $color-white;
    transition: all $transition-speed ease-in;
    border: 1px solid $color-block-border;
    width: calc(100% - 3rem);

    &:hover,
    &:focus {
      border-color: $color-block-border;
      outline: none;
      box-shadow: 1px 1px 3px $color-block-border;
    }
  }

  &__note {
    color: $color-text-gray;
    font-size: 0.8rem;
    margin: 1rem 0 0;
    display: block;
    text-align: left;
  }
}
