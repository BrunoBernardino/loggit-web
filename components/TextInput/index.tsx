import React, { forwardRef } from 'react';
import styled from 'styled-components';

import './style.scss';

interface TextInputProps {
  label?: string;
  name: string;
  placeholder?: string;
  isDisabled?: boolean;
  type?: 'text' | 'email' | 'password' | 'url';
  value: any;
  onChange?: any;
  onKeyDown?: any;
  onClick?: any;
  className?: string;
  ref?: any;
  id?: string;
  note?: string;
  title?: string;
}

const StyledInput = styled.input``;

const TextInput: React.FC<TextInputProps> = forwardRef(
  (props: TextInputProps, ref: any) => {
    const {
      label,
      id,
      name,
      className,
      isDisabled,
      note,
      ...remainingProps
    } = props;

    return (
      <div className={`${className || ''} TextInput`}>
        {label ? (
          <label className="TextInput__label" htmlFor={id || name}>
            {label}
          </label>
        ) : null}
        <StyledInput
          className="TextInput__input"
          disabled={isDisabled}
          id={id || name}
          name={name}
          ref={ref}
          {...remainingProps}
        />
        {note ? <p className="TextInput__note">{note}</p> : null}
      </div>
    );
  },
);

TextInput.defaultProps = {
  isDisabled: false,
  type: 'text',
};

export default TextInput;
