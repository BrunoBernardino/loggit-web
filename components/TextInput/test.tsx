import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import expect from 'expect';
import enzymify from 'expect-enzyme';

import TextInput from './index';

expect.extend(enzymify());

describe('TextInput', () => {
  // TODO: enzyme still doesn't work with React 17: https://github.com/enzymejs/enzyme/issues/2429
  it.skip('renders the text input without errors', () => {
    const wrapper = mount(
      <TextInput label="Name" name="name" value="Bruno" onChange={() => {}} />,
    );
    expect(wrapper.find('input').props().value).toBe('Bruno');
    expect(wrapper.find('label').text()).toBe('Name');
  });

  it('renders the input as expected', () => {
    const tree = renderer
      .create(<TextInput label="Name" name="name" value="Bruno" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
