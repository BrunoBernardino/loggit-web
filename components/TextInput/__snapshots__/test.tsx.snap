// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders the input as expected 1`] = `
<div
  className=" TextInput"
>
  <label
    className="TextInput__label"
    htmlFor="name"
  >
    Name
  </label>
  <input
    className="TextInput__StyledInput-sc-1v53buj-0 TextInput__input"
    disabled={false}
    id="name"
    name="name"
    type="text"
    value="Bruno"
  />
</div>
`;
